Fixes and updates to check_paths

cassella opened the pull request:
When check_paths was updated for the master->main change, the github
url we look for was updated but the length to remove from the token
wasn't.  Fix that (44->42).

Add a few more written-out alternatives to be ignored.

Update location and extension for the adding-a-tasking-model and
comm-layer documents, and add new special exceptions for
adding-a-launcher, SpellChecking, and RuntimeLibrary.

Add a few symbols (<, >, **) to the list to split tokens on.

---
Signed-off-by: Paul Cassella <gitgitgit@manetheren.bigw.org>

gbtitus commented on util/devel/check_paths:
```diff
@@ -39,9 +39,9 @@ def check_path(p):
-    if p == "compiler/module":
+    if (p == "compiler/runtime" or p == "compiler/module" or
+        p == "runtime/library" or p == "runtime/lib" or
+        p == "module/files" or p == "test/directory" ):
```
Think we want "modules/files" (plural "modules"), at least if that string is for this one:
```
doc/rst/developer/chips/inactive/21.rst:74: modules/files not found
```

gbtitus commented on util/devel/check_paths:
```diff
@@ -109,22 +109,34 @@ def check_token(t):
             return True
+    elif filename == "doc/rst/developer/implementation/adding-a-launcher.rst":
+        # And this one's <name>s
+        if "<name>" in token:
```
If we're now splitting on '<' and '>' (in `check_line()`) does this case actually occur?

gbtitus commented on util/devel/check_paths:
```diff
@@ -109,22 +109,34 @@ def check_token(t):
+    elif filename == "doc/rst/developer/bestPractices/RuntimeLibrary.rst":
+        if "MORE-DIRS" in token:
+            return True
+
```
Should we add an exception, by which I mean hack, for this one?
```
doc/rst/platforms/cray.rst:196: util/setchplenv not found
```

gbtitus reviewed (changes requested):
I've marked it "Request changes" only because of my first comment, which I think is a spot that isn't quite right. The second is something I think may be incorrect but could reflect my own lack of understanding. And the third is more an indication that I understand this sort of thing can get out of hand ... 😁

gbtitus commented:
@cassella Did you see the feedback here?

gbtitus commented:
I threw the "post release" label on here, as an advisory.

cassella commented:
I added those fixes, and another change resulting from #19513.

(I added that one as a todo in the file rather than an issue because 1. It doesn't seem anyone but me runs this script anyway, and 2. I assume you don't want some passer-by picking it up and asking for help from the group with something that doesn't affect anyone but me.)

gbtitus reviewed (approved):
👍

gbtitus commented:
I'll merge this as soon as we post-releasely thaw ...

ronawho commented:
> It doesn't seem anyone but me runs this script anyway, and 2. I assume you don't want some passer-by picking it up and asking for help from the group with something that doesn't affect anyone but me

Do you think this is something we should be running more often or as part of CI testing? Or does this feel more like spellcheck where we want to run this occasionally, but don't want to hold up PRs over it? (Ideally I think run occasionally would be like a semi-automated job run, but unfortunately is currently whenever Paul is kind enough to)

gbtitus commented:
> Do you think this is something we should be running more often or as part of CI testing?

I'm not positive I'm the referent of "you" in your question but my own opinion is that until the output is solidly free of false positives and we're all on board with updating the script to prevent new ones as they arise, no, we shouldn't do it as part of CI testing. But running it as we do spellcheck would be okay. I do think it's valuable, it's just that it requires maintenance effort if it's to be really worthwhile.

ronawho commented:
> I'm not positive I'm the referent of "you" in your question 

Was directed at Paul

> But running it as we do spellcheck would be okay.

Right, be we (anybody other than Paul) don't really run spellcheck today. I don't think we have a good workflow/process for these types of checks today.

cassella commented:
Those are good ideas, thanks!  I'll push those changes presently.

Alas there are still too many false positives, so it's still a manual process to run it.

cassella commented:
I named the function `remove_prefix` since that's closer to the name of the function coming in python 3.9.
